be/src/olap/compaction.cpp

```diff
@@ -79,7 +79,10 @@ bool is_rowset_tidy(std::string& pre_max_key, const RowsetSharedPtr& rhs) {
     // check segment size
     auto* beta_rowset = reinterpret_cast<BetaRowset*>(rhs.get());
     std::vector<size_t> segments_size;
-    static_cast<void>(beta_rowset->get_segments_size(&segments_size));
+    Status status = beta_rowset->get_segments_size(&segments_size);
```

Comment:
for usage like this, you can introduce a new marco like `RETURN_FALSE_IF_ERROR` to deal with them